### PR TITLE
Disable Go Workspaces when executing 'go list' in packaging process

### PR DESCRIPTION
**Pull request by AndersonQ**
running 'go list' with Go worspaces enabled will report all modules listed on go.work, what would make GetModuleName to fail as it requires 'go list' to return only one module, the agent itself.

## Proposed commit message

>Disable Go Workspaces when executing 'go list' in packaging process
>
>running 'go list' with Go worspaces enabled will report all modules listed on go.work, what would make GetModuleName to fail as it requires 'go list' to return only one module, the agent itself.

## What does this PR do?

Disable Go Workspaces when executing 'go list' in agent packaging process

## Why is it important?

Running 'go list' with Go worspaces enabled will report all modules listed on go.work, what would make GetModuleName to fail as it requires 'go list' to return only one module, the agent itself.
Therefore if Go workspaces is enabled, packaging the agent will fail

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- ~~[ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)~~
- ~~[ ] I have added an integration test or an E2E test~~

## How to test this PR locally

 - create a Go workspace
 - add a few modules, e.g. `elastic-agent-client`, `elastic-agent-libs`
 - run `mage package`
 - without this change it'll fail

## Related issues

- N/A

## Questions to ask yourself

- How are we going to support this in production? 
- How are we going to measure its adoption? 
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -419,3 +419,5 @@ replace (
 
 // Exclude this version because the version has an invalid checksum.
 exclude github.com/docker/distribution v2.8.0+incompatible
+
+replace github.com/elastic/elastic-agent-client/v7 => github.com/AndersonQ/elastic-agent-client/v7 v7.0.0-20240109062623-907af5c552fb
```
Comment (rdner) [inline go.mod:423]: Did you leave it on purpose? I don't think it's a good idea.

Comment (AndersonQ) [reply go.mod:423]: thanks, my bad

Comment (belimawr) [reply go.mod:423]: @AndersonQ it seems you haven't pushed the update for this line.

**Review by belimawr — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (AndersonQ) [reply go.mod]: done

**Review by belimawr — APPROVED** [review]

**Review by rdner — APPROVED** [review]

**Pull request merged**